### PR TITLE
Add utility to configure ECN WRED parameters

### PR DESCRIPTION
**Pull request by andriymoroz-mlnx**
Signed-off-by: Andriy Moroz <c_andriym@mellanox.com>

Should be merged AFTER https://github.com/Azure/sonic-buildimage/pull/1178

### CODE REVIEW

**Comment by lguohan** [comment]
@yxieca , to review

File: config/main.py [hunk 1]
```diff
@@ -331,6 +331,30 @@ def incremental(file_name):
...
+@click.option('-v', '--verbose', is_flag=True, help="Enable verbose output")
+def ecn(profile, rmax, rmin, ymax, ymin, gmax, gmin, verbose):
+    """ECN-related configuration tasks"""
+    command = "ecnconfig -p %s" % profile
+    if rmax: command += " -rmax %d" % rmax
+    if rmin: command += " -rmin %d" % rmin
+    if ymax: command += " -ymax %d" % ymax
+    if ymin: command += " -ymin %d" % ymin
+    if gmax: command += " -gmax %d" % gmax
+    if gmin: command += " -gmin %d" % gmin
```
Comment (yxieca) [inline config/main.py]: The way you tested these parameters will ignore a parameter when it is 0 or None. Is this what you intended?

File: config/main.py [hunk 2]
```diff
@@ -331,6 +331,30 @@ def incremental(file_name):
...
+    if verbose:
+        click.echo(command)
```
Comment (yxieca) [inline config/main.py]: Do you think you want to further pass verbose option to ecnconfig script?

File: scripts/ecnconfig [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+                ecn.set_wred_threshold(args.profile, "gmax", args.green_max)
+            if args.green_min:
+                ecn.set_wred_threshold(args.profile, "gmin", args.green_min)
+            if args.yellow_max:
+                ecn.set_wred_threshold(args.profile, "ymax", args.yellow_max)
+            if args.yellow_min:
+                ecn.set_wred_threshold(args.profile, "ymin", args.yellow_min)
+            if args.red_max:
+                ecn.set_wred_threshold(args.profile, "rmax", args.red_max)
+            if args.red_min:
```
Comment (yxieca) [inline scripts/ecnconfig]: All above if statements in try block will ignore value 0 and None, is that intentional?

Comment (yxieca) [inline scripts/ecnconfig]: Maybe deleting comments is not the best thing.

I figured out that args are returning values in strings, so we don't have a risk of ignoring integer value 0 here.

File: config/main.py (hunk 1 above)
Comment (andriymoroz-mlnx) [reply config/main.py]: yes, min/max parameters are optional

File: config/main.py (hunk 2 above)
Comment (andriymoroz-mlnx) [reply config/main.py]: right, will add

File: config/main.py (hunk 1 above)
Comment (yxieca) [reply config/main.py]: Sorry for the late reply. Andriy,

My question is more about if you intentionally ignores the value 0? I understand you would like to ignore when it is None.

Regards,
Ying

**Pull request merged**